honeybadger/payload.py

```diff
@@ -54,14 +54,28 @@ def read_source(frame, source_radius=3):
     return {}
 
 def server_payload(config):
-    return {
+    payload =  {
         'project_root': config.project_root,
         'environment_name': config.environment,
         'hostname': config.hostname,
         'time': datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ"),
         'pid': os.getpid(),
         'stats': stats_payload()
     }
+    if config.is_aws_lambda_environment:
```

Comment:
The data you're collecting here looks correct, but the payload isn't quite right. We could have documented #60 better—sorry about that. :)

Here's a link to how the Ruby gem's Lambda plugin currently updates the notice payload:

https://github.com/honeybadger-io/honeybadger-ruby/blob/master/lib/honeybadger/plugins/lambda.rb#L13

I believe `data` on line 13 is the same data as you're collecting here. That needs to be moved to a new top-level key in the JSON payload: `payload['details']['Lambda Details'] = data` in `create_payload` [here](https://github.com/honeybadger-io/honeybadger-python/blob/87bce097f3197f6888c805cd125c8d45996d65de/honeybadger/payload.py#L113). Here's a [JSON Schema](https://github.com/honeybadger-io/honeybadger-schemas/blob/master/exceptions/v1/notices.json#L423) that describes the notice payload.

While you're making this change, could you also set `notice['request']['component']`, `notice['request']['action']`, and `notice['request']['context']['lambda_trace_id']`, per the Ruby gem example I linked to above?

Reply:
if i update the payload from `create_payload` method, it'll end up in the `request` key which i believe is meant for details associated with the request. Technically shouldn't those data be part of the server payload which is the lambda in this case?
Or do you mean add "details" as a new key(same level with error, server & request)